src/zenml/zen_stores/migrations/versions/12eff0206201_rename_workspace_to_project.py

```diff
@@ -0,0 +1,768 @@
+        batch_op.alter_column(
+            "project_id",
+            existing_type=sqlmodel.sql.sqltypes.GUID(),
+            nullable=False,
+        )
+        batch_op.drop_constraint(
+            "fk_service_workspace_id_workspace", type_="foreignkey"
+        )
+        batch_op.create_foreign_key(
+            "fk_service_project_id_project",
+            "project",
+            ["project_id"],
+            ["id"],
+            ondelete="CASCADE",
+        )
+        batch_op.drop_column("workspace_id")
+
+    # step_run table
+    with op.batch_alter_table("step_run", schema=None) as batch_op:
+        # Phase 1: Add new column as nullable
+        batch_op.add_column(
+            sa.Column(
+                "project_id", sqlmodel.sql.sqltypes.GUID(), nullable=True
+            )
+        )
+
+    # Phase 2: Copy data
+    op.execute(
+        """
+        UPDATE `step_run` 
+        SET project_id = workspace_id
+        """
+    )
+
+    # Phase 3: Add constraints and drop old column
+    with op.batch_alter_table("step_run", schema=None) as batch_op:
+        batch_op.alter_column(
+            "project_id",
+            existing_type=sqlmodel.sql.sqltypes.GUID(),
+            nullable=False,
+        )
+        batch_op.drop_constraint(
+            "fk_step_run_workspace_id_workspace", type_="foreignkey"
+        )
+        batch_op.create_foreign_key(
+            "fk_step_run_project_id_project",
+            "project",
+            ["project_id"],
+            ["id"],
+            ondelete="CASCADE",
+        )
+        batch_op.drop_column("workspace_id")
+
+    # trigger table
+    with op.batch_alter_table("trigger", schema=None) as batch_op:
+        # Phase 1: Add new column as nullable
+        batch_op.add_column(
+            sa.Column(
+                "project_id", sqlmodel.sql.sqltypes.GUID(), nullable=True
+            )
+        )
+
+    # Phase 2: Copy data
+    op.execute(
+        """
+        UPDATE `trigger` 
+        SET project_id = workspace_id
+        """
+    )
+
+    # Phase 3: Add constraints and drop old column
+    with op.batch_alter_table("trigger", schema=None) as batch_op:
+        batch_op.alter_column(
+            "project_id",
+            existing_type=sqlmodel.sql.sqltypes.GUID(),
+            nullable=False,
+        )
+        batch_op.drop_constraint(
+            "unique_trigger_name_in_workspace", type_="unique"
+        )
+        batch_op.create_unique_constraint(
+            "unique_trigger_name_in_project", ["name", "project_id"]
+        )
+        batch_op.drop_constraint(
+            "fk_trigger_workspace_id_workspace", type_="foreignkey"
+        )
+        batch_op.create_foreign_key(
+            "fk_trigger_project_id_project",
+            "project",
+            ["project_id"],
+            ["id"],
+            ondelete="CASCADE",
+        )
+        batch_op.drop_column("workspace_id")
+
+    with op.batch_alter_table("user", schema=None) as batch_op:
```

Comment:
We drop this but don't seem to add a `default_project_id` as replacement?

Reply:
You're right. I don't really need this anymore. It was meant as a temporary measure for a quicker PoC, but now we decided to go with a fully updated dashboard which supports creating/switching between workspaces, so I dropped it.

Reply:
I thought we maybe wanted to even have that as an option for the user to configure in the dashboard, to essentially select which project will be active by default each time they access a workspace. But I guess when we get to that point we can always implement that